language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -381,12 +381,15 @@ public static Position toPosition(LinePosition linePosition) {
                 TypeSymbol errorType = CommonUtil.getRawType(((ErrorTypeSymbol) rawType).detailTypeDescriptor());
                 StringBuilder errorString = new StringBuilder("error (\"\"");
                 if (errorType.typeKind() == TypeDescKind.RECORD) {
```

Comment:
Shall we add the max depth validation as well? If not there is a chance of SO for the following

```
type ER1 error<record {| ER1 e?; |}>;

public function main() {
    ER1 er1 = getER1();
}

function getER1() {
return error("", e<cursor>)
```